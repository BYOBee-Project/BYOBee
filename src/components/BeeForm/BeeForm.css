.bee-form {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 10vh;
}

.bee-form-title {
  font-family: 'Londrina Solid', cursive;
  font-size: 5vh;
}

.bee-form-form {
  width: 90%;
  display: flex;
  flex-direction: column;
  align-items: center;
  border-radius: 5px;
  background-color: hsla(4, 82%, 67%, 0.827);
  border-radius: 10px;
  padding: 2vh 4vw 2vh 4vw;
  gap: 0.5vh;
}

.bee-form-label {
  margin: 5px;
  font-weight: bolder;
  text-align: center;
  margin-top: 1vh;
  font-family: 'Londrina Solid', cursive;
}

.bee-form-button {
  padding-top: 0.5vh;
  padding-bottom: 0.5vh;
  margin-top: 2vh;
}
