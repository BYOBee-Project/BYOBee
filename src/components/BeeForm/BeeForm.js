import React, { useState } from 'react';
import './BeeForm.css';
import { useEffect } from 'react';
import { findBeeById } from '../../services/bees';
import { useParams } from 'react-router-dom';

export default function BeeForm({ currentUser }) {
  const params = useParams();
  const [newBee, setNewBee] = useState('');
  const [date, setDate] = useState('');
  const [photo, setPhoto] = useState('');
  const [observation, setObservation] = useState('');
  const [location, setLocation] = useState('');

  useEffect(() => {
    const fetchBee = async () => {
      const data = await findBeeById(params.id);
      setNewBee(data);
    };
    fetchBee();
  }, [params.id]);

  const handleSubmit = (e) => {
    e.preventDefault();
    console.log('clicking');
  };

  return (
    <div className="BeeForm">
      <h1>Add a {newBee.name} to your collection!</h1>
      <form className="bee-form-form">
        <label className="bee-form-label">Add a photo:</label>
        <input type="file" />
        <label className="bee-form-label">Add a date:</label>
        <input
          type="date"
          onChange={(e) => {
            setDate(e.target.value);
          }}
        />
        {/* when we have location capabilities, update this */}
        <label className="bee-form-label">Where did you encounter the bee?</label>
        <input
          type="textfield"
          onChange={(e) => {
            setLocation(e.target.value);
          }}
        />
        <label className="bee-form-label">Any other notes or observations?</label>
        <input
          type="textarea"
          onChange={(e) => {
            setObservation(e.target.value);
          }}
        />
        <button onClick={handleSubmit}>Save</button>
      </form>
    </div>
  );
}
