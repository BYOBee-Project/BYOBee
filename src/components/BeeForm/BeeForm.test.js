import { render, screen } from '@testing-library/react';
import BeeForm from './BeeForm.js';
import { MemoryRouter } from 'react-router-dom';

test('About page should render about us stuff', async () => {
  const container = render(
    <MemoryRouter>
      <BeeForm newBee={{ newBee: 'Prunus Miner Bee' }} />
    </MemoryRouter>
  );
  screen.findByText('Add a photo:');
  expect(container).toMatchSnapshot();
});
