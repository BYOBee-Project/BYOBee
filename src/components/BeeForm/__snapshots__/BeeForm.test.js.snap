// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About page should render about us stuff 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bee-form"
      >
        <h1
          class="bee-form-title"
        />
        <form
          class="bee-form-form"
        >
          <label
            class="bee-form-label"
          >
            Add a photo:
          </label>
          <input
            accept="image/*"
            id="single"
            required=""
            type="file"
          />
          <label
            class="bee-form-label"
          >
            Add a date:
          </label>
          <input
            required=""
            type="date"
            value=""
          />
          <label
            class="bee-form-label"
          >
            Where did you encounter the bee?
          </label>
          <input
            required=""
            type="textfield"
            value=""
          />
          <label
            class="bee-form-label"
          >
            Any other notes or observations?
          </label>
          <input
            required=""
            type="textarea"
            value=""
          />
          <button
            class="bee-form-button"
          >
            Save
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bee-form"
    >
      <h1
        class="bee-form-title"
      />
      <form
        class="bee-form-form"
      >
        <label
          class="bee-form-label"
        >
          Add a photo:
        </label>
        <input
          accept="image/*"
          id="single"
          required=""
          type="file"
        />
        <label
          class="bee-form-label"
        >
          Add a date:
        </label>
        <input
          required=""
          type="date"
          value=""
        />
        <label
          class="bee-form-label"
        >
          Where did you encounter the bee?
        </label>
        <input
          required=""
          type="textfield"
          value=""
        />
        <label
          class="bee-form-label"
        >
          Any other notes or observations?
        </label>
        <input
          required=""
          type="textarea"
          value=""
        />
        <button
          class="bee-form-button"
        >
          Save
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
